client/src/components/QA/AnswerModal.jsx

```diff
@@ -27,13 +27,68 @@ class AnswerModal extends React.Component {
       email:'',
       answerError: '',
       nicknameError: '',
-      emailError: ''
+      emailError: '',
+      answerPass: false,
+      nicknamePass: false,
+      emailPass: false,
+      images:[],
+      uploadImgBtn: true
     };
     this.handleSubmitError = this.handleSubmitError.bind(this);
+    this.handleImageUpload = this.handleImageUpload.bind(this);
+    this.handleInput = this.handleInput.bind(this);
   };
 
+  handleInput(e) {
+    var type = e.target.id;
+    var value = e.target.value;
+    if (type === 'Email') {
```

Comment:
If you coerce your type to be lower-case and use an [includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) to be sure it's a valid "type" and then set state directly
```
validTypes.includes(type) ? 
this.setState({[type]: value}) : this.setState({error: {type: "validation", message: `incorrect type, ${type}`}})`